lib/ansible/galaxy/collection/galaxy_api_proxy.py

```diff
@@ -28,11 +28,16 @@
 class MultiGalaxyAPIProxy:
     """A proxy that abstracts talking to multiple Galaxy instances."""
 
-    def __init__(self, apis, concrete_artifacts_manager):
-        # type: (t.Iterable[GalaxyAPI], ConcreteArtifactsManager) -> None
+    def __init__(self, apis, concrete_artifacts_manager, offline=False):
+        # type: (t.Iterable[GalaxyAPI], ConcreteArtifactsManager, bool) -> None
         """Initialize the target APIs list."""
         self._apis = apis
         self._concrete_art_mgr = concrete_artifacts_manager
+        self._offline = offline  # Prevent all GalaxyAPI calls
+
+    def _offline_is_unsupported(self):
+        if self._offline:
+            raise NotImplementedError
```

Comment:
Let's be more verbose about what's happening here:
```suggestion
            raise NotImplementedError("The calling code is not supposed to be invoked in 'offline' mode.")
```

Reply:
Done